rules.d/99-systemd.rules.in

```diff
@@ -22,7 +22,9 @@ SUBSYSTEM=="block", TAG+="systemd"
 SUBSYSTEM=="block", ENV{DM_SUSPENDED}=="1", IMPORT{db}="SYSTEMD_READY"
 SUBSYSTEM=="block", ENV{DM_SUSPENDED}=="1", GOTO="systemd_end"
 
-SUBSYSTEM=="block", ACTION=="add", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", ENV{SYSTEMD_READY}="0"
+# DM_UDEV_DISABLE_OTHER_RULES_FLAG==1 means that the device shouldn't be probed.
+# Import previous SYSTEMD_READY state.
+SUBSYSTEM=="block", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", ENV{SYSTEMD_READY}=="", IMPORT{db}="SYSTEMD_READY"
```

Comment:
> So, clean rules in our codebase should never have IMPORT{db}, i.e. we want that uevents stand on their own, and come to the same conclusion always. 

That isn't possible for dm. There are situations in which the device can't be probed, while their "ready" status shouldn't change. We can't set `SYSTEMD_READY=0` in this situation, because that might cause umounts or similar destructive actions on systemd's part. The current rule set has been amended multiple times to avoid this kind of issue.

> By looking at the db they have a major input from the side, so that properties stick around forever, if they ever were set incorrectly.

I see your point, but that can't happen here, because the import statement is bound to `DM_UDEV_DISABLE_OTHER_RULES_FLAG`. This flag indicates a temporary condition that will usually be reverted by the next uevent. It is true that one must be carefule with `IMPORT{db}`. If you look at my [dm patch set](https://gitlab.com/lvmteam/lvm2/-/merge_requests/9), you'll see that I have removed a couple of cases where properties were mistakenly imported.

> If you look at the rules we ship, we never ise IMORT{db} ourselves, with the exception of the DM support code, this here. If you touch this drop the whole concept please, and determine anew whether the device is ready or not.

Like I said, we can't. The best we can do, if you insist, is to move the `IMPORT{db}="SYSTEMD_READY"` statement into the dm ruleset. But it would still need to be done in the systemd rule set as well, otherwise the systemd rules would yield wrong results with all but the latest device-mapper rules.